hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1139,7 +1131,7 @@ public boolean scheduleLogCompactionAtInstant(String instantTime, Option<Map<Str
    * @return Collection of WriteStatus to inspect errors and counts
    */
   public HoodieWriteMetadata<O> logCompact(String logCompactionInstantTime) {
-    return logCompact(logCompactionInstantTime, config.shouldAutoCommit());
+    return logCompact(logCompactionInstantTime, false);
```

Comment:
Does the semantics change because the `config.shouldAutoCommit()` was by default true before?

Reply:
nope. all user facing writers were using auto commit disabled flow except for few exceptions. 
- Spark data source writes 
- spark sql 
-spark streaming
- HoodieStreamer 
- Flink 

Exceptions: 
bootstrap and PartitionTTL management job uses auto commit enabled flow. 
So, we have introduced "INTERNAL_AUTO_COMMIT" config to be used in theses. For user of Hudi, there is not going to be any auto commit flows hereafter. 
unless someone was using Writeclient directly, this should not have any side effects. 
For someone who has directly written any code using WriteClient and if they were relying on auto commit flows, when they upgrade to 1.1, they might have to fix their code to call writeClient.commit() everywhere.

Reply:
Should we remove the auto commit completely? I saw we rename the original option to add "internal" keyword, but it still looks hacky. It looks like we used this option to do auto-commit in action executors just like before.

Reply:
There are 2 use-cases where are relying on auto commit flow. 
bootstrap and partition TTL management. 
if you check the places in source code where we enable the new internal auto commit flow, you will realize.

Reply:
I will try to take another stab to see if we can get rid of it completely.